### PR TITLE
JIT ARM64 SVE: Add FZ_2A, HG_2A, GZ_3A, GV_3A, GY_3*, DV_4A

### PR DESCRIPTION
**Pull request by amanasifkhalid**
Part of #94549. Implements the following encodings:

* IF_SVE_FZ_2A
* IF_SVE_HG_2A (SVE2)
* IF_SVE_GZ_3A
* IF_SVE_GV_3A
* IF_SVE_GY_3B_D (SVE2)
* IF_SVE_GY_3A (SVE2)
* IF_SVE_DV_4A (SVE2)

cstool output:
```
sqcvtn        z0.h, { z2.s, z3.s }
sqcvtun       z6.h, { z14.s, z15.s }
uqcvtn        z14.h, { z30.s, z31.s }
bfmlalb       z0.s, z1.h, z0.h[0]
bfmlalt       z2.s, z3.h, z1.h[1]
bfmlslb       z4.s, z5.h, z2.h[2]
bfmlslt       z6.s, z7.h, z3.h[3]
fmlalb        z8.s, z9.h, z4.h[4]
fmlalt        z10.s, z11.h, z5.h[5]
fmlslb        z12.s, z13.h, z6.h[6]
fmlslt        z14.s, z15.h, z7.h[7]
fcmla z0.s, z1.s, z0.s[0], #0
fcmla z2.s, z3.s, z5.s[1], #90
fcmla z4.s, z5.s, z10.s[0], #180
fcmla z6.s, z7.s, z15.s[1], #270
```

JitDisasm output:
```
sqcvtn  z0.h, { z2.s, z3.s }
sqcvtun z6.h, { z14.s, z15.s }
uqcvtn  z14.h, { z30.s, z31.s }
bfmlalb z0.s, z1.h, z0.h[0]
bfmlalt z2.s, z3.h, z1.h[1]
bfmlslb z4.s, z5.h, z2.h[2]
bfmlslt z6.s, z7.h, z3.h[3]
fmlalb  z8.s, z9.h, z4.h[4]
fmlalt  z10.s, z11.h, z5.h[5]
fmlslb  z12.s, z13.h, z6.h[6]
fmlslt  z14.s, z15.h, z7.h[7]
fcmla   z0.s, z1.s, z0.s[0], #0
fcmla   z2.s, z3.s, z5.s[1], #90
fcmla   z4.s, z5.s, z10.s[0], #180
fcmla   z6.s, z7.s, z15.s[1], #270
```

### CODE REVIEW

**Comment by amanasifkhalid** [comment]
cc @dotnet/arm64-contrib

File: src/coreclr/jit/emitarm64.cpp [hunk 1]
```diff
@@ -12099,7 +12171,7 @@ void emitter::emitIns_R_R_R_I(instruction ins,
             break;
 
         case INS_sve_bfmul:
-            assert(opt = INS_OPTS_SCALABLE_H);
+            assert(opt == INS_OPTS_SCALABLE_H);
```
Comment (kunalspathak) [inline src/coreclr/jit/emitarm64.cpp:12174]: good catch

File: src/coreclr/jit/emitarm64.cpp [hunk 2]
```diff
@@ -22654,6 +22889,21 @@ void emitter::emitDispLowPredicateRegPair(regNumber reg, insOpts opt)
     printf(" }, ");
 }
 
+//------------------------------------------------------------------------
+// emitDispVectorRegPair: Display a pair of vector registers
+//
+void emitter::emitDispVectorRegPair(regNumber reg, insOpts opt)
```
Comment (kunalspathak) [inline src/coreclr/jit/emitarm64.cpp:22895]: can we reuse `emitDispSveConsecutiveRegList` instead?

Comment (amanasifkhalid) [reply src/coreclr/jit/emitarm64.cpp]: Yes we can; updated.

File: src/coreclr/jit/emitarm64.cpp [hunk 3]
```diff
@@ -24825,6 +25090,17 @@ void emitter::emitDispInsHelp(
             emitDispReg(id->idReg3(), id->idOpSize(), false);                                   // mmmmm
             break;
 
+        // <Pd>, <Pn>, <Pm>.<T>[<Wv>, <imm>]
+        case IF_SVE_DV_4A: // ........ix.xxxvv ..NNNN.MMMM.DDDD -- SVE broadcast predicate element
+            emitDispPredicateReg(id->idReg1(), PREDICATE_NONE, id->idInsOpt(), true);   // DDDD
```
Comment (a74nh) [inline src/coreclr/jit/emitarm64.cpp:25095]: These should use the get predicate function instead of `PREDICATE_*`

Comment (amanasifkhalid) [reply src/coreclr/jit/emitarm64.cpp:25095]: Got it, fixed.

**Review by a74nh — APPROVED** [review]
I'm happy with this now.

**Review by TIHan — APPROVED** [review]
LGTM

Hopefully when capstone gets updated this month, we will be able to decode the unsupported ones.

**Pull request merged**